rclcpp/include/rclcpp/copy_all_parameters.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Open Navigation LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__COPY_ALL_PARAMETERS_HPP_
+#define RCLCPP__COPY_ALL_PARAMETERS_HPP_
+
+#include <vector>
+#include <string>
+#include "rcl_interfaces/srv/list_parameters.hpp"
+#include "rclcpp/parameter.hpp"
+
+namespace rclcpp
+{
+
+/**
+ * @brief A method to copy all parameters from one node (parent) to another (child).
+ * May throw parameter exceptions in error conditions
+ * @param parent Node to copy parameters from
+ * @param child Node to copy parameters to
+ */
```

Comment:
style nitpick:

```suggestion
/// Copy all parameters from one source node to another destination node.
/**
 * May throw parameter exceptions in error conditions
 * \param source Node to copy parameters from
 * \param destination Node to copy parameters to
 */
```

We use `///` for briefs, and the `\` escape key. Also, I think source and destination are probably better terms. Parent/child implies an on going relationship between the nodes, at least to me, which isn't the case.